.travis.yml

```diff
@@ -74,31 +85,68 @@ install:
     # other dependencies.
 
 matrix:
-    include:
 
-        # Do a coverage test in Python 3.
-        - python: 3.6
+    # Don't wait for allowed failures
+    fast_finish: true
+
+    include:
+        # Make sure that egg_info works without dependencies
+        - stage: Initial tests
+          env: PYTHON_VERSION=3.7 SETUP_CMD='egg_info'
+
+        # Try MacOS X, usually enough only to run from cron as hardly there are
+        # issues that are not picked up by a linux worker
+        - os: osx
+          stage: Cron tests
+          env: SETUP_CMD='test' EVENT_TYPE='cron'
+
+        # Do a coverage test.
+        - os: linux
+          stage: Initial tests
           env: SETUP_CMD='test --coverage'
 
         # Check for sphinx doc build warnings - we do this first because it
         # may run for a long time
-        - python: 3.6
-          env: SETUP_CMD='build_sphinx -w'
+        - os: linux
+          env: SETUP_CMD='build_docs -w'
+               PIP_DEPENDENCIES="$PIP_DEPENDENCIES sphinx-astropy"
```

Comment:
```suggestion
               PIP_DEPENDENCIES="`echo $PIP_DEPENDENCIES sphinx-astropy`"
```

Reply:
(as this has been moved to one of your branches I coundn't push the change, but this should solve the issue)

Reply:
I'm looking at this, too, right now. I think I had multiple error at the same time. The reason that this is so difficult to track down is that I had two "matrix" definitions in the travis.yml. Apparently, I copied in the new one from some other file and forgot to delete the old one. Travis actually executed the old block, while we have been debugging the new block for most of the time. 

That's something I've finally found. Now I will add you fix (or go back to conda for sphinx-astropy) and it's probably all work and I'll wonder why it was so difficult... Thanks for helping out!

Reply:
`sphinx-astropy` is not conda packaged. And even if it were, for CI, stick with pip, it's quicker and more efficient. 

The two matrix section should not have been a problem, the first nested one one should be looped through all the language settings while the second one adds individual jobs to it. Cleaning up is great though.

Reply:
I thought it's conda packaged here: https://github.com/astropy/conda-channel-astropy/blob/master/requirements.yml#L128

Reply:
yes, you're right: https://anaconda.org/astropy/repo?type=conda&label=main

somehow it didn't show up in my local conda search. Sorry.